osquery/logger/plugins/aws_util.cpp

```diff
@@ -179,10 +191,46 @@ OsqueryFlagsAWSCredentialsProvider::GetAWSCredentials() {
                                    FLAGS_aws_secret_access_key);
 }
 
-OsqueryAWSCredentialsProviderChain::OsqueryAWSCredentialsProviderChain()
+Aws::Auth::AWSCredentials
+OsquerySTSAWSCredentialsProvider::GetAWSCredentials() {
+  // Grab system time in seconds since epoch for token experation checks
+  auto now = std::chrono::system_clock::now();
+  auto epoch = now.time_since_epoch();
+  auto seconds = std::chrono::duration_cast<std::chrono::seconds>(epoch);
+  // Pull new STS creds if we dont have them cached from a previous run,
+  // otherwise return cached STS creds
+  if (token_expire_time <= seconds.count()) {
+    // Create and setup a STS client to pull our temp creds
+    VLOG(1) << "Generate new AWS STS credentials.";
+    initAwsSdk();
+    Status s = makeAWSClient<Aws::STS::STSClient>(sts_client_, false);
+    Aws::STS::Model::AssumeRoleRequest sts_r;
+    sts_r.SetRoleArn(FLAGS_aws_sts_arn_role);
+    sts_r.SetRoleSessionName(FLAGS_aws_sts_session_name);
+    sts_r.SetDurationSeconds(FLAGS_aws_sts_timeout);
+    // Pull our STS creds
+    auto sts_outcome = sts_client_->AssumeRole(sts_r);
```

Comment:
I know these are nasty templated types, but I would prefer to see them (rather than `auto`) so it's easier to look up the docs in the future.